### PR TITLE
Issue 773: add support to rate limiter spring instance configuration …

### PR DESCRIPTION
**Pull request by Romeh**
…customizer

### CODE REVIEW

File: resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * the composite  of any circuit breaker {@link RateLimiterConfigCustomizer} implementations.
```
Comment (dlsrb6342) [inline resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java]: Please fix typo
```suggestion
 * the composite  of any rate limiter {@link RateLimiterConfigCustomizer} implementations.
```

File: resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class CompositeRateLimiterCustomizer {
+
+    private final Map<String, RateLimiterConfigCustomizer> customizerMap = new HashMap<>();
+
+    public CompositeRateLimiterCustomizer(List<RateLimiterConfigCustomizer> customizers) {
+        if (customizers != null && !customizers.isEmpty()) {
+            customizerMap.putAll(customizers.stream()
+                .collect(
+                    Collectors.toMap(RateLimiterConfigCustomizer::name, Function.identity())));
```
Comment (dlsrb6342) [inline resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java]: Don't we need to check that there are multiple customizer for same name?

Comment (Romeh) [reply resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java]: we can throw an exception to show that is not allowed to have more than one customizer per instance name , is this what u r asking for :) ? or i misunderstood ?

**Comment by Romeh** [comment]
> I think that if there is a method that users can add or replace Customizer after `Composite**Customizer` initialized, it would be good. What do you think?

can u please describe more when the user need to do that?  or what will be the use case ?

File: resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java (hunk 2 above)
Comment (dlsrb6342) [reply resilience4j-framework-common/src/main/java/io/github/resilience4j/common/ratelimiter/configuration/CompositeRateLimiterCustomizer.java]: Yes right. We can throw an exception or write on docs about supporting just one customizer per instance name.

**Comment by dlsrb6342** [comment]
> > I think that if there is a method that users can add or replace Customizer after `Composite**Customizer` initialized, it would be good. What do you think?
> 
> can u please describe more when the user need to do that? or what will be the use case ?

Sorry. I misunderstood that when customizer is applied. Please never mind this comment.

**Pull request merged**